pkg/statedb/types.go

```diff
@@ -93,7 +108,18 @@ type RWTable[Obj any] interface {
 	// iteration of updates and deletions (see (*DeleteTracker[Obj]).Process).
 	Delete(WriteTxn, Obj) (oldObj Obj, hadOld bool, err error)
 
-	// DeleteAll removes all objects in the table.
+	// CompareAndSwap compares the existing object's revision against the
+	// given revision and if equal it deletes the object.
+	//
+	// Possible errors:
+	// - ErrRevisionNotEqual: the object has mismatching revision
+	// - ErrObjectNotFound: object not found from the table
```

Comment:
Does it make sense for `ErrObjectNotFound` to be an error for `CompareAndDelete`? I get the use-case for `ErrRevisionNotEqual`, but in symmetry with the normal `Delete`, I'd have expected that `CompareAndDelete` suceeds when the object is gone. 

EDIT: I realise now that `Delete` also doesn't do this. Shouldn't Delete be an idempotent operation?

Reply:
Ah but I wanted it to be symmetric with `CompareAndSwap`, e.g. the `Compare` part of it failed with `ErrObjectNotFound`. But I can see it working the other way around with being symmetric to `Delete`.

Reply:
After bit of a think, let's make it symmetric with `Delete` and drop the error. Most use-cases likely are not interested in the `ErrObjectNotFound` in the first place and they can just check if an object was returned or not.